nebula-logger/main/log-management/lwc/loggerSettings/loggerSettings.js

```diff
@@ -0,0 +1,370 @@
+/*************************************************************************************************
+ * This file is part of the Nebula Logger project, released under the MIT License.               *
+ * See LICENSE file or go to https://github.com/jongpie/NebulaLogger for full license details.   *
+ ************************************************************************************************/
+
+// UI
+import { LightningElement, wire } from 'lwc';
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+
+// LoggerSettings__c metadata
+import LOGGER_SETTINGS_OBJECT from '@salesforce/schema/LoggerSettings__c';
+import { getObjectInfo } from 'lightning/uiObjectInfoApi';
+import canUserModifyLoggerSettings from '@salesforce/apex/LoggerSettingsController.canUserModifyLoggerSettings';
+import getLoggingLevelOptions from '@salesforce/apex/LoggerSettingsController.getLoggingLevelOptions';
+import getSaveMethodOptions from '@salesforce/apex/LoggerSettingsController.getSaveMethodOptions';
+import getSetupOwnerTypeOptions from '@salesforce/apex/LoggerSettingsController.getSetupOwnerTypeOptions';
+import getShareAccessLevelOptions from '@salesforce/apex/LoggerSettingsController.getShareAccessLevelOptions';
+
+// LoggerSettings__c data
+import getOrganization from '@salesforce/apex/LoggerSettingsController.getOrganization';
+import getRecords from '@salesforce/apex/LoggerSettingsController.getRecords';
+import createNewRecord from '@salesforce/apex/LoggerSettingsController.createNewRecord';
+import saveRecord from '@salesforce/apex/LoggerSettingsController.saveRecord';
+import deleteRecord from '@salesforce/apex/LoggerSettingsController.deleteRecord';
+import searchForSetupOwner from '@salesforce/apex/LoggerSettingsController.searchForSetupOwner';
+
+export default class LoggerSettings extends LightningElement {
+    // UI
+    title = 'Logger Settings';
+    columns;
+    isReadOnlyMode = true;
+    showLoadingSpinner = false;
+    showEditModal = false;
+    showDeleteModal = false;
+    isNewOrganizationRecord = false;
+    showSetupOwnerLookup = false;
+    showSetupOwnerDropdown = false;
+    setupOwnerSearchTerm;
+    setupOwnerSearchResults;
+    showDropdown = false;
+    showPill = false;
+
+    // LoggerSettings__c metadata
+    canUserModifyLoggerSettings;
+    loggerSettingsFields;
+    loggingLevelOptions;
+    saveMethodOptions;
+    setupOwnerTypeOptions;
+    shareAccessLevelOptions;
+
+    // LoggerSettings__c data
+    records;
+    currentRecord;
+    selectedSetupOwner;
+
+    connectedCallback() {
+        document.title = this.title;
+        this.showLoadingSpinner = true;
+
+        getOrganization().then(result => {
```

Comment:
slight optimization possible here - you could wrap each of your `thennable` promises and then call `Promise.all` on an array of those promises so that they can run in parallel.

Reply:
Ooooh, that's a great idea - I'll make that change ASAP. Thanks for taking a look!

Reply:
I've optimized this in [this commit](https://github.com/jongpie/NebulaLogger/pull/254/commits/8d48fa11d8d1996849f67727593669d363bc2152) by consolidating all of the different Apex methods into a single method that returns all of the various picklist options in a single object

Reply:
Makes sense - and as you said, what a speed up!